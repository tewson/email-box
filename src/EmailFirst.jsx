import juice from 'juice';
import PropTypes from 'prop-types';
import React from 'react';
import { renderToStaticMarkup } from 'react-dom/server';

const EmailFirst = ({ children, css }) => {
  const inlined = juice(
    `<style>${css}</style>${renderToStaticMarkup(children)}`,
  );

  return (
    <div
      className="email-first"
      style={{
        maxWidth: 600,
      }}
      dangerouslySetInnerHTML={{ __html: inlined }}
    />
  );
};

EmailFirst.propTypes = {
  children: PropTypes.node.isRequired,
  css: PropTypes.string.isRequired,
};

export default EmailFirst;
