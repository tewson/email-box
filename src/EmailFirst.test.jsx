/* eslint-env jest */

import React from 'react';
import { render, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import EmailFirst from './EmailFirst';

describe('EmailFirst', () => {
  it('should be defined', () => {
    expect(EmailFirst).toBeDefined();
  });

  it('should render correctly', () => {
    expect(toJson(shallow(<EmailFirst />))).toMatchSnapshot();
  });

  it('should render children correctly', () => {
    expect(
      toJson(
        shallow(
          <EmailFirst>
            <div>
              <h1>Test</h1>
              <p>Children</p>
            </div>
          </EmailFirst>,
        ),
      ),
    ).toMatchSnapshot();

    const randomChild = <p>{Math.random()}</p>;

    expect(shallow(<EmailFirst>{randomChild}</EmailFirst>).html()).toContain(
      shallow(randomChild).html(),
    );
  });

  it('should inline css', () => {
    const css = `
      .irrelevant-class {
        color: red;
      }

      .relevant-class {
        color: green;
      }

      .nested-element {
        color: blue;
      }
    `;

    const wrapper = render(
      <EmailFirst css={css}>
        <div className="relevant-class">
          <h1 className="nested-element">Nested Element</h1>
        </div>
      </EmailFirst>,
    );

    const targetDiv = wrapper.find('.relevant-class');

    expect(targetDiv.attr('style')).toContain('color: green');
    expect(targetDiv.css('color')).toEqual('green');

    const targetNestedElement = wrapper.find('.nested-element');

    expect(targetNestedElement.attr('style')).toContain('color: blue');
    expect(targetNestedElement.css('color')).toEqual('blue');
  });
});
