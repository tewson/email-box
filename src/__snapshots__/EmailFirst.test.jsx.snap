// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailFirst should render children correctly 1`] = `
<div
  className="email-first"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<div><h1>Test</h1><p>Children</p></div>",
    }
  }
  style={
    Object {
      "maxWidth": 600,
    }
  }
/>
`;

exports[`EmailFirst should render correctly 1`] = `
<div
  className="email-first"
  dangerouslySetInnerHTML={
    Object {
      "__html": "",
    }
  }
  style={
    Object {
      "maxWidth": 600,
    }
  }
/>
`;
